#! /bin/sh /usr/share/dpatch/dpatch-run
## 15_gtk_dropdown.dpatch by  <glandium@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix for focus problem with drop down lists. bz#281551

@DPATCH@

--- xulrunner/widget/src/gtk2/nsWindow.cpp	(revision 168)
+++ xulrunner/widget/src/gtk2/nsWindow.cpp	(local)
@@ -485,10 +485,8 @@
 nsWindow::IsVisible(PRBool & aState)
 {
     aState = mIsVisible;
-    if (mIsTopLevel && mShell && !GTK_WIDGET_MAPPED(mShell)) {
-        /* we do not change mIsVisible to PR_FALSE here so we don't bother
-           to change it back to PR_TRUE when the mShell is mapped again. */
-        aState = PR_FALSE;
+    if (mIsTopLevel && mShell) {
+        aState = GTK_WIDGET_VISIBLE(mShell);
     }
     return NS_OK;
 }
